compiler/src/iree/compiler/Utils/ElementPackingUtils.cpp

```diff
@@ -72,52 +82,14 @@ Value calculateStorageElementCountInBytes(Location loc,
     staticCount *= IREE::Util::getRoundedElementByteWidth(alignedElementType);
   }
 
-  // TODO: Do we use makeComposedFoldedAffineApply here, so the index
-  // computation an be much simpler.
-  SmallVector<int64_t> paddedShape(shapedType.getShape());
-  SmallVector<Value> paddedDynamicDims(dynamicDims.begin(), dynamicDims.end());
-  auto encoding = IREE::Encoding::getEncodingAttr(shapedType);
-  if (encoding && !encoding.getRoundDimsToArray().empty()) {
-    auto roundDimsTo = encoding.getRoundDimsToArray();
-    FailureOr<linalg::ContractionDimensions> cDims =
-        IREE::Encoding::getEncodingContractionDims(encoding);
-    auto pad = [&](int dim, int value) {
-      std::optional<unsigned> maybeMappedDim =
-          encoding.mapDimToOperandIndex(dim);
-      if (!maybeMappedDim) {
-        return;
-      }
-      unsigned mappedDim = maybeMappedDim.value();
-      if (shapedType.isDynamicDim(mappedDim)) {
-        auto alignment = builder.create<arith::ConstantIndexOp>(loc, value);
-        paddedDynamicDims[mappedDim] = builder.create<arith::CeilDivUIOp>(
-            loc, paddedDynamicDims[mappedDim], alignment);
-        paddedDynamicDims[mappedDim] = builder.create<arith::MulIOp>(
-            loc, paddedDynamicDims[mappedDim], alignment);
-      } else {
-        paddedShape[mappedDim] = llvm::alignTo(paddedShape[mappedDim], value);
-      }
-    };
-    for (auto m : cDims->m) {
-      pad(m, roundDimsTo[0]);
-    }
-    for (auto n : cDims->n) {
-      pad(n, roundDimsTo[1]);
-    }
-    for (auto k : cDims->k) {
-      pad(k, roundDimsTo[2]);
-    }
-  }
-
   for (unsigned i = 0; i < shapedType.getRank(); ++i) {
```

Comment:
After the change, are we able to guarantee here that we don't need to pad shape?